### ISSUE TITLE
Improve logging across the package

### ISSUE THREAD

**Issue by Gonmeso** [gizatechxyz/giza-agents#56]
We should add multiple debug logs to help the developers to debug users problems.

Some debug logs that could be added:

* log requests and responses
* log contract initialization
* log ape actions
* framework args serialization
* input and output of osiris des/serialization

This should be done using the logging module and we should create a utility function that can easily be added to start the log level of the library like:

```python
from giza.agents.utils import debug_logging

# This should make logging to show all the DEBUG and INFO logs
debug_logging()
```

**Comment by Josh-121** [comment]
can I work on this?

**Comment by Gonmeso** [comment]
Fixed on #60
